app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -86,6 +86,7 @@ def complete
 
     cookies["_bypass_cache"] = true
     cookies[:authentication_data] = { value: client_hash.to_json, path: Discourse.base_path("/") }
+    secure_session[:oauth] = true
```

Comment:
Seems like all other uses of `secure_session` are using string keys; might not be a big deal but probably best to be consistent